syncer/malsync/syncer.go

```diff
@@ -361,13 +361,13 @@ func (s *Syncer) downloadMalfeasanceProofs(ctx context.Context, initial bool, up
 			sst.done()
 			if initial && sst.numSyncedPeers() >= s.cfg.MinSyncPeers {
 				if err := s.updateState(); err != nil {
-					return nil
+					return err
```

Comment:
Perhaps the failure cases deserve tests, wdyt?

Reply:
Well, it is possible, but on the other hand if we're to cover all the database access errors that would be rather large amount of tests. And when an SQL request fails, it's usually something very very wrong (such as out of disk space or database corruption) and `go-spacemesh` will fail badly in any case, and in multiple places